src/main/java/com/puppycrawl/tools/checkstyle/DefaultLogger.java

```diff
@@ -211,4 +214,93 @@ private void closeStreams() {
         }
     }
 
+    /**
+     * Represents a message that can be localised. The translations come from
+     * message.properties files. The underlying implementation uses
+     * java.text.MessageFormat.
+     */
+    private static final class LocalizedMessage {
+
+        /**
+         * A cache that maps bundle names to ResourceBundles.
+         * Avoids repetitive calls to ResourceBundle.getBundle().
+         */
+        private static final Map<String, ResourceBundle> BUNDLE_CACHE =
+                Collections.synchronizedMap(new HashMap<>());
+
+        /**
+         * The locale to localise messages to.
+         **/
+        private static Locale locale = Locale.getDefault();
+
+        /**
+         * Key for the message format.
+         **/
+        private final String key;
+
+        /**
+         * Arguments for MessageFormat.
+         */
+        private final Object[] args;
+
+        /**
+         * Creates a new {@code LocalizedMessage} instance.
+         *
+         * @param key the key to locate the translation.
+         */
+        /* package */ LocalizedMessage(String key) {
+            this.key = key;
+            args = null;
+        }
+
+        /**
+         * Creates a new {@code LocalizedMessage} instance.
+         *
+         * @param args arguments for the translation.
+         * @param key the key to locate the translation.
+         */
+        /* package */ LocalizedMessage(Object[] args, String key) {
```

Comment:
make it `LocalizedMessage(String key, Object[] args)`

{main target} , {how to change target}

Reply:
Please read https://github.com/checkstyle/checkstyle/pull/9904#discussion_r646078818

Reply:
replied, still need to be fixed.

Reply:
Done